src/components/inputs/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from "react";
+import styled from "@emotion/styled";
+import { css, Theme } from "@emotion/react";
+import CheckIcon from "components/inputs/TextInput/CheckIcon";
+
+interface Props {
+  label?: string;
+  placeholder?: string;
+  conditionList?: string[];
+  conditionChecks?: ((text: string) => boolean)[];
+  multiline?: boolean;
+  errorMessage?: string;
+}
+
+const TextInput: React.FC<Props> = ({
+  label,
+  placeholder = "",
+  conditionList,
+  conditionChecks,
+  multiline = false,
+  errorMessage = "잘못된 입력입니다.",
+}) => {
+  const [value, setValue] = useState("");
+  const [isValid, setIsValid] = useState(true);
+  const [isFocused, setIsFocused] = useState(false);
+  const [borderStyle, setBorderStyle] = useState("default");
+
+  const handleFocus = () => {
+    setIsFocused(true);
+  };
+
+  const handleBlur = () => {
+    setIsFocused(false);
+    setIsValid(conditionChecks ? conditionChecks.every((check) => check(value)) : true);
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const newValue = event.target.value;
+    setValue(newValue);
+
+    // 입력된 텍스트의 조건 확인을 수행하여 isValid 상태 업데이트
+    if (isFocused) {
+      setIsValid(conditionChecks ? conditionChecks.every((check) => check(newValue)) : true);
+    }
+  };
```

Comment:
포커싱이 되면 특정 로직이 실행되는 것은 좋은 것 같습니다. 그러나, 이렇게 처리할 경우 처음부터 무조건 빨간 박스가 뜰 것 같습니다. (물론 비밀번호 등의 경우 이렇게 처음부터 보안강도 validation 을 하는 곳도 있으나, ) 모든 input 에 대해 클릭하자마자 빨간색으로 표시되는 것은 UX 에 조금 좋지 않을 것 같습니다 🙂 

최초로 focus 했을 때는 validate 하지 않고, 최초 focus 후 blur 이벤트가 1회라고 발생했다면 validate 하는 로직은 어떨까요~?

Reply:
```suggestion
   // 입력된 텍스트의 조건 확인을 수행하여 isValid 상태 업데이트
    if (isFocused && isBlurred && value != "") {
      setIsValid(conditionCheckList ? conditionCheckList.every((check) => check(newValue)) : true);
    }
  };
```
Blur 이벤트 최초 발생 후에 validate 하도록 수정했습니다!

Reply:
넵 수정 감사합니다~~